### DISCUSSION TITLE
How many years of forward compatibility can be expected?

### DISCUSSION THREAD

**Discussion by Emily511511** [AppImage/AppImageKit#1347] [Q&A] [answered]
If an appimage is properly packaged without taking an unrealistic amount of space, how many years of forward compatibility can be expected **for abandonware closed source software** packaged in an appimage on all major distros?

**Comment by probonopd** [comment, marked as answer]
Hi @Emily511511, great question!

If you would like to maximize forward compatibility, then you should make sure to use a static AppImage runtime, and bundle _all_ dependencies, down to and including glibc, in your AppImage (something e.g., go-appimage `appimagetool -s deploy` can help doing). This way, the AppImage should work until the kernel ABI interface breaks (very, very unlikely as long as Linus steers the kernel), or future versions of `fusermount` become incompatible, at which point in time you'd need to repackage the contents of the AppImage (a very straightforward process).

So, how long? Nobody knows for sure. But a few decades should hopefully be possible. And if you archive a Linux Live ISO alongside it to be run in a future emulator, a few hundred years should hopefully be realistic.

I am happy with helping you with this exciting project.

**Comment by Emily511511** [comment]
Hey @probonopd

Thank you a lot for your answer, it was very helpful 🙂.

It's nice to know that you are willing, but I don't plan on building an appimage yet 😅

But still, thanks for your reply!
